sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -621,12 +621,23 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       ds.filter(_ => true),
       Some(1), Some(2), Some(3))
   }
+
+  test("SPARK-13540 add Scala object to OuterScopes") {
+    OuterScopes.addOuterScope(OuterObject)
```

Comment:
Oh, at first I got a wrong fix for this issue (see the first commit).  I thought the problem was that when adding the Scala object into `OuterScopes`, the class name without the `$` suffix should be added. Later I found I was wrong, but forgot to remove this line. Thanks!